pkg/lib/cockpit-components-install-dialog.jsx

```diff
@@ -207,3 +218,56 @@
     check_missing();
     return prom;
 }
+
+function install_dialog_immutable(pkg, options) {
+    return new Promise((resolve, reject) => {
+        const missing_name = <strong>{pkg.join(", ")}</strong>;
+
+        const body = {
+            id: "dialog",
+            title: options.immutable_title || _("Add software"),
+            body: (
+                <div className="scroll">
+                    <p>{ format_to_fragments(options.immutable_text || _("$0 needs to be added to the OS image."),
+                                             missing_name) }</p>
+                </div>
+            ),
+        };
+
+        const footer = {
+            actions: [],
+            cancel_button: { text: _("Close"), variant: "secondary" },
+            dialog_done: reject
+        };
+
+        show_modal_dialog(body, footer);
+    });
+}
+
+function install_dialog_error(pkg, details) {
+    return new Promise((resolve, reject) => {
+        const missing_name = <strong>{pkg.join(", ")}</strong>;
+
+        const body = {
+            id: "dialog",
+            title: _("Installation error"),
+            titleIconVariant: "danger",
+            body: (
+                <div className="scroll">
+                    <p>{ format_to_fragments(_("The $0 package is missing but can not be installed."),
+                                             missing_name) }</p>
+                    <p>{ format_to_fragments(_("Contacting the PackageKit service returned an error: $0"),
+                                             details.error) }</p>
+                </div>
+            ),
+        };
+
+        const footer = {
+            actions: [],
+            cancel_button: { text: _("Close"), variant: "secondary" },
+            dialog_done: reject
+        };
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18240-20230127-135755-f796be5b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)